docs/synapse/userguides/storm_ref_filter.rstorm

```diff
@@ -793,247 +921,521 @@ know the exact order or values of the array itself.
 ::
-    
-    <query> +#cno.infra.dns.sinkhole@=(2017/01/01, 2018/01/01)
+  
+  <query> +#rep.*.**2017**
+
+The tag glob filter above uses both a single and double asterisk. The single asterisk matches any tag element
+in the second position; the double asterisk matches any string that includes "2017", including across "dot"
+boundaries. The filter will match all of the following:
+
+- ``rep.alienvault.cve20178291``
+- ``rep.malwarebazaar.3p.reversinglabs.document_ole_exploit_cve_2017_11882``
+- ``rep.vt.cve_2017_0199``
+
+.. NOTE::
+  
+  The double asterisk must match "something" - the filter above matches strings where "2017" appears
+  **between** other arbitrary characters. The filter would not match tags such as ``rep.foo.2017`` or
+  ``rep.bar.baz.cve2017``.
+
+
+.. _filter-tag-timestamp:
+
+Filter Using Tag Timestamp Values
++++++++++++++++++++++++++++++++++
+
+A tag timestamp can be thought of as a specialized "property" of a tag that happens to be a date / time range
+(interval). You can filter nodes based on tag timestamp values using any comparison operator supported by interval
+(:ref:`type-ival` types). The time / interval extended operator ( ``@=`` ) is used most often, but equal to
+( ``=`` ) can also be used to **exactly** match the values in the interval.
+
+See :ref:`filter-interval` for additional detail on the use of the ``@=`` operator.
+
 
+**Syntax:**
 
-- Filter results to exclude nodes associated with threat cluster 17 after January 1, 2019:
+*<query>* **+** | **-** **#** *<tag>* **@=** *<time>* | **(** *<min_time>* **,** *<max_time>* **)**
 
+*Filter the current result set to only include nodes that were associated with anonymous VPN infrastructure between December 1, 2023 and January 1, 2024:*
 
-.. storm-pre:: inet:fqdn -#cno.threat.t17.own@=(2019/01/01, now)
+.. storm-pre:: [ (inet:fqdn=wazn.airservers.org +#cno.infra.anon.vpn=(2023/05/24 23:16:51.423, 2023/12/05 23:12:40.626)) (inet:ipv6=2607:9000:0:85:68a3:75b4:13ab:770a +#cno.infra.anon.vpn=(2023/08/15 00:12:15,2023/12/05 23:12:54) ) ] +#cno.infra.anon.vpn @= ( 2023/12/01, 2024/01/01 )
 ::
-    
-    <query> -#cno.threat.t17.own@=(2019/01/01, now)
+  
+  <query> +#cno.infra.anon.vpn @= ( 2023/12/01, 2024/01/01 )
+
+*Filter the current working set to only include nodes that were owned / controlled by Threat Cluster 15 as of October 30, 2021:*
+
+.. storm-pre:: [ inet:fqdn=ronthecat.com +#cno.threat.t15.own=(2020/09/12, 2022/09/12) ] +#cno.threat.t15.own @= 2021/10/30
+::
+  
+  <query> +#cno.threat.t15.own @= 2021/10/30
+
+.. NOTE::
+  
+  While it is possible to filter based on "tags that have timestamps" or "tags that do not have timestamps",
+  it currently requires more advanced Storm syntax. For example, the following filter will **include** only
+  those nodes with the tag ``cno.threat.t15.own`` that **do not** have timestamps (``.0=$lib.null``):
+  
+  ``+$((#cno.threat.t15.own).0=$lib.null)``
```

Comment:
This actually needs one more part to it for the "check if it has the cno.threat.t15.own tag at all". The minus filter doesn't need this check.
```suggestion
  ``+$(#cno.threat.15.own and (#cno.threat.t15.own).0=$lib.null)``
```

Reply:
Fixed (again). TIL! :-)

Reply:
Unfixed / removed :laughing: